genesyscloud/resource_genesyscloud_user.go

```diff
@@ -1231,21 +1274,41 @@ func updateUserProfileSkills(d *schema.ResourceData, usersAPI *platformclientv2.
 	return nil
 }
 
-func updateUserRoutingUtilization(d *schema.ResourceData, usersAPI *platformclientv2.UsersApi) diag.Diagnostics {
+func updateUserRoutingUtilization(d *schema.ResourceData, usersAPI *platformclientv2.UsersApi, sdkConfig *platformclientv2.Configuration) diag.Diagnostics {
 	if d.HasChange("routing_utilization") {
 		if utilConfig := d.Get("routing_utilization").([]interface{}); utilConfig != nil {
+			var err error
+
+			log.Printf("Updating user utilization")
+
 			if len(utilConfig) > 0 { // Specified but empty utilization list will reset to org-wide defaults
-				sdkSettings := make(map[string]platformclientv2.Mediautilization)
+				// Update settings
 				allSettings := utilConfig[0].(map[string]interface{})
-				for sdkType, schemaType := range utilizationMediaTypes {
-					if mediaSettings, ok := allSettings[schemaType]; ok && len(mediaSettings.([]interface{})) > 0 {
-						sdkSettings[sdkType] = buildSdkMediaUtilization(mediaSettings.([]interface{}))
+				labelUtilizations := allSettings["label_utilizations"].([]interface{})
+
+				if labelUtilizations != nil && len(labelUtilizations) > 0 {
+					routingAPI := platformclientv2.NewRoutingApiWithConfig(sdkConfig)
+					apiClient := &routingAPI.Configuration.APIClient
+
+					path := fmt.Sprintf("%s/api/v2/routing/users/%s/utilization", routingAPI.Configuration.BasePath, d.Id())
+					headerParams := buildHeaderParams(routingAPI)
+					requestPayload := make(map[string]interface{})
+					requestPayload["utilization"] = buildMediaTypeUtilizations(allSettings)
+					requestPayload["labelUtilizations"] = buildLabelUtilizationsRequest(labelUtilizations)
+					_, err = apiClient.CallAPI(path, "PUT", requestPayload, headerParams, nil, nil, "", nil)
```

Comment:
does routingAPI not have direct apimethod to make this call

Reply:
No, as mentioned on the description and the previous PR https://github.com/MyPureCloud/terraform-provider-genesyscloud/pull/793.

The new field, labelUtilizations, is hidden in the API, which means it doesn't show up in the routingAPI, and won't show up, until the feature is released.

Once it is, I'll put up another PR removing the direct call, calling routingAPI instead.